finite_difference/src/field_mod.f90

```diff
@@ -1040,7 +1040,7 @@ subroutine halo_exchange(self, depth)
     ! Locals
     integer :: exch  !> Handle for exchange
 
-    call exchange_generic(b2=self%data, handle=exch, &
+    call exchange_generic(b2=self%get_data(), handle=exch, &
```

Comment:
Please could you similarly update the docstring for this routine to say that data is first copied to the host (if necessary).

Reply:
Done